src/test/scala/io/iohk/ethereum/domain/BlockchainSpec.scala

```diff
@@ -44,6 +44,15 @@ class BlockchainSpec extends AnyFlatSpec with Matchers with ScalaCheckPropertyCh
     assert(validBlock == block.get)
   }
 
+  it should "be able to do strict check of block existence in the chain" in new EphemBlockchainTestSetup {
+    val validBlock = Fixtures.Blocks.ValidBlock.block
+    blockchain.save(validBlock, Seq.empty, ChainWeight(100, 100), saveAsBestBlock = true)
+    assert(blockchain.isInChain(validBlock.hash))
+    // simulation of node restart
+    blockchain.saveBestKnownBlocks(validBlock.header.number - 1)
+    assert(!blockchain.isInChain(validBlock.hash))
```

Comment:
curious: why not use scala test matchers?

Reply:
there are a bunch of tests like this, will change them